openpype/hosts/houdini/plugins/publish/collect_vray_rop.py

```diff
@@ -43,6 +43,26 @@ def process(self, instance):
         render_products = []
         # TODO: add render elements if render element
 
+        # Store whether we are splitting the render job in an export + render
+        # TODO: check names of VRay parms
+        export_job = bool(rop.parm("render_export_mode").eval())
+        instance.data["exportJob"] = export_job
+        export_prefix = None
+        export_products = []
+        if export_job:
+            export_prefix = evalParmNoFrame(
+                rop, "render_export_filepath", pad_character="0"
+            )
+            beauty_export_product = self.get_render_product_name(
```

Comment:
There is no `get_render_product_name` function introduced, do you mean `get_beauty_render_product` in the code?(Maybe we should rename a little bit)

Reply:
good call, all the other `collect` plugins call this function `get_render_product_name` so I can rename it to that instead so they are all more aligned. However, this brings the question to whether we would maybe even want to abstract that function from all collect plugins and reuse it on all, there's no need to make it a class function really so it could go to `houdini.api.lib` or somewhere else

**Vray**
```python
    def get_render_product_name(self, prefix, suffix="<reName>"):
        """Return the beauty output filename if render element enabled
        """
        aov_parm = ".{}".format(suffix)
        beauty_product = None
        if aov_parm in prefix:
            beauty_product = prefix.replace(aov_parm, "")
        else:
            beauty_product = prefix

        return beauty_product
````

**Mantra**
```python
    def get_render_product_name(self, prefix, suffix):
        product_name = prefix
        if suffix:
            # Add ".{suffix}" before the extension
            prefix_base, ext = os.path.splitext(prefix)
            product_name = prefix_base + "." + suffix + ext

        return product_name
```
**Karma**
```python
    def get_render_product_name(self, prefix, suffix):
        product_name = prefix
        if suffix:
            # Add ".{suffix}" before the extension
            prefix_base, ext = os.path.splitext(prefix)
            product_name = "{}.{}{}".format(prefix_base, suffix, ext)

        return product_name
```

**Arnold**
```python
    def get_render_product_name(self, prefix, suffix):
        """Return the output filename using the AOV prefix and suffix"""

        # When AOV is explicitly defined in prefix we just swap it out
        # directly with the AOV suffix to embed it.
        # Note: ${AOV} seems to be evaluated in the parameter as %AOV%
        if "%AOV%" in prefix:
            # It seems that when some special separator characters are present
            # before the %AOV% token that Redshift will secretly remove it if
            # there is no suffix for the current product, for example:
            # foo_%AOV% -> foo.exr
            pattern = "%AOV%" if suffix else "[._-]?%AOV%"
            product_name = re.sub(pattern,
                                  suffix,
                                  prefix,
                                  flags=re.IGNORECASE)
        else:
            if suffix:
                # Add ".{suffix}" before the extension
                prefix_base, ext = os.path.splitext(prefix)
                product_name = prefix_base + "." + suffix + ext
            else:
                product_name = prefix

        return product_name
```

**Redshift**
```python
    def get_render_product_name(self, prefix, suffix):
        """Return the output filename using the AOV prefix and suffix"""

        # When AOV is explicitly defined in prefix we just swap it out
        # directly with the AOV suffix to embed it.
        # Note: ${AOV} seems to be evaluated in the parameter as %AOV%
        has_aov_in_prefix = "%AOV%" in prefix
        if has_aov_in_prefix:
            # It seems that when some special separator characters are present
            # before the %AOV% token that Redshift will secretly remove it if
            # there is no suffix for the current product, for example:
            # foo_%AOV% -> foo.exr
            pattern = "%AOV%" if suffix else "[._-]?%AOV%"
            product_name = re.sub(pattern, suffix, prefix, flags=re.IGNORECASE)
        else:
            if suffix:
                # Add ".{suffix}" before the extension
                prefix_base, ext = os.path.splitext(prefix)
                product_name = prefix_base + "." + suffix + ext
            else:
                product_name = prefix

        return product_name
```